shotover-proxy/tests/opensearch_int_tests/mod.rs

```diff
@@ -1,52 +1,242 @@
+use crate::shotover_process;
 use opensearch::{
     auth::Credentials,
     cert::CertificateValidation,
-    http::Url,
+    http::response::Response,
     http::{
-        response::Response,
         transport::{SingleNodeConnectionPool, TransportBuilder},
-        StatusCode,
+        Method, StatusCode, Url,
     },
-    indices::IndicesExistsParts,
+    indices::{IndicesCreateParts, IndicesDeleteParts, IndicesExistsParts},
     params::Refresh,
-    BulkOperation, BulkParts, OpenSearch, SearchParts,
+    BulkOperation, BulkParts, DeleteParts, Error, IndexParts, OpenSearch, SearchParts,
 };
 use serde_json::{json, Value};
 use test_helpers::docker_compose::docker_compose;
 
-pub async fn index_documents(client: &OpenSearch) -> Response {
-    let index = "posts";
-    let exists_response = client
-        .indices()
-        .exists(IndicesExistsParts::Index(&[index]))
-        .send()
-        .await
-        .unwrap();
+async fn assert_ok_and_get_json(response: Result<Response, Error>) -> Value {
+    let response = response.unwrap();
+    let status = response.status_code();
 
-    assert_eq!(exists_response.status_code(), StatusCode::NOT_FOUND);
+    if response.method() == Method::Head {
+        if status != StatusCode::OK {
+            panic!("Opensearch HEAD query returned status code {status}");
+        }
+        Value::Null
+    } else {
+        let json = response.json().await.unwrap();
```

Comment:
We always get a valid json even if the status is not ok? Shouldn't we unwrap after testing the response status?

Reply:
Yep, opensearch returns valid json even when the http code is not successful.
We include the json in our panic message because it includes a detailed error message.